common/urlbuilder/url.go

```diff
@@ -17,6 +19,7 @@ var ErrInvalidURL = errors.New("URL format is incorrect")
 type URL struct {
 	delegate           *url.URL
 	queryParams        url.Values
+	unEncodeParams     map[string]struct{}
```

Comment:
please modify to `unencodedQueryParams        url.Values`, and add comments.

(See my message below to understand why `url.Values` makes sense as the type.